python/ray/serve/tests/test_pipeline_driver.py

```diff
@@ -100,5 +101,25 @@ def test_dag_driver_custom_schema(serve_instance):
     assert resp.json() == 100
 
 
+@serve.deployment
+def combine(*args):
+    return list(args)
+
+
+def test_dag_driver_partial_input(serve_instance):
+    with InputNode() as inp:
+        dag = DAGDriver.bind(
+            combine.bind(echo.bind(inp[0]), echo.bind(inp[1]), echo.bind(inp[2])),
+            input_schema=json_request,
+        )
+    handle = serve.run(dag)
+    assert ray.get(handle.predict.remote([1, 2, [3, 4]])) == [1, 2, [3, 4]]
```

Comment:
sorry i actually meant `handle.predict.remote(1, 2, [3, 4])` so it's multiple args that will spawn `InputAttributeNode`   rather than dealing with one python object. You can see them in https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/experimental/dag/tests/test_input_node.py?L225

Reply:
This doesn't make sense for typical DAGDriver here. The input is still a single item because predict and http request has a single value representing the input. As you can see in the dag users can still write input[0] to index into input node.

Reply:
how about we use the same internal data wrapper as ray dag's execute ? https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/experimental/dag/input_node.py?L255 

I understand eventually InputNode translate to one python object, but it's better if we can minimize the churn from user's dev flow if they find out they need to change input syntax / structure from ray core to serve dag. I'm not suggesting what we have now is already optimal, but they should be the same class.